Lib/fontbakery/specifications/googlefonts.py

```diff
@@ -245,8 +245,8 @@ def check_DESCRIPTION_min_length(fb, descfile):
 
 @register_condition
 @condition
-def metadata(font):
-  family_dir = os.path.dirname(font)
+def metadata(fonts):
+  family_dir = os.path.dirname(fonts[0])
```

Comment:
Sure! Semantically a `family_directory` condition would make sense.

> safe to use fonts[0] 

unless fonts is empty

> halt with a critical error if the user passes an empty set of TTF files

yeah, if we can work with less assumptions it's better. Also, if you make such an assumption, that a list won't be empty, write at least a comment or documentation for it.

> I think it is better to check for that right in the beginning ... instead of having to check for it again everywhere else.

Yes and no. Tough, a personal TODO of me somewhere on my disk says: 

```py

  #TODO:
  #
  # * Add a way to check for missing arguments or wrong types in the
  #   testrunner `values` property (via spec).
  # * Add custom user input validation to the spec, to check
  #   values brought in via testrunners `values`.
```

It's still important to remember that too much auto-magic makes code unreadable.

Also, it would still require us to document that we don't expect e.g. an empty list, just in case someone wants to reuse a test in a personal specification (which maybe wouldn't have the same input validation).

Reusability across specs is already hard because we define condition names just as we want, without thinking about a bigger world outside, but for that we could write another wrapper layer ;-)

> ... (in the testsuite runner script)

That's why I put a big disclaimer at the top of all of these modules "don't make this domain specific" :-D